lib/blockchain/chaindb.js

```diff
@@ -1868,9 +1885,8 @@ class ChainDB {
     if (this.options.spv)
       return undefined;
 
-    // Write actual block data (this may be
-    // better suited to flat files in the future).
-    this.put(layout.b.encode(hash), block.encode());
+    // Write actual block data
+    this.blocksBatch.writeBlock(hash, block.encode());
```

Comment:
I like how you are preserving the batch API, but since most of our blockstore writes will be single blocks, I wonder if it's worth the overhead to create `new FileBatch` and `new WriteOp`? I guess those objects are pretty minimal but in a few years bootstrapping nodes will have 100,000s of these to write.

Reply:
Without preserving batch API the way logic works becomes significantly harder to follow. You can check bcoin for that, it comes to the point that `removeBlock` method in bcoin does not remove the block and it becomes responsibility of the caller.

So I believe performance hit is negligible compared to the things that happen around that method.